pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -944,6 +969,57 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 	}
 }
 
+func (c *Controller) initNamespaceWatch(opts Options) {
+	var err error
+	c.nsWatch, err = c.client.CoreV1().Namespaces().Watch(context.TODO(), metav1.ListOptions{FieldSelector: "metadata.name=" + opts.SystemNamespace})
```

Comment:
I am not sure watch should be used. I think informers handle a bunch of stuff around syncing, reconnects, etc: https://stackoverflow.com/questions/59544139/kubernetes-client-go-watch-interface-vs-cache-newinformer-vs-cache-newsharedi#:~:text=Informers%20is%20a%20higher%20level,and%20an%20in%2Dmemory%20cache.

Reply:
might want similar to https://github.com/istio/istio/blob/8bb59d16bd5697f5510fdde13b8afd4b8af7b859/pkg/kube/configmapwatcher/configmapwatcher.go#L52 - this could probably actually be made generic (although might require some casting due to lack of real generics)

Reply:
I think the init is all we really want; I think the kube controller's internal queue should process the updates.